plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/utils/JdkFetcher.java

```diff
@@ -0,0 +1,270 @@
+package io.jenkins.tools.pluginmodernizer.core.utils;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A utility class for fetching the latest JDK releases from the Adoptium GitHub repository.
+ */
+@SuppressFBWarnings(value = "PATH_TRAVERSAL_IN", justification = "false, positive")
+public class JdkFetcher {
+    private static final Logger LOG = LoggerFactory.getLogger(JdkFetcher.class);
+
+    private static final String GITHUB_API_URL = "https://api.github.com/repos/adoptium";
+    private static final String DOWNLOAD_DIR = System.getProperty("user.home") + "/.jdks/";
```

Comment:
What if the user.home is not writable ? Why we don't rely on cache dir to specify an writable location